### PR TITLE
feat: Support import and export of categories

### PR DESCRIPTION
**Pull request by johan-bjareholt**
Todo:
- [x] Better design
- [x] Finish import

@ErikBjare Do you think that the categorization has a good format right now that we should proceed with, or are there any improvements you have in mind before essentially declaring it a standard format?

### CODE REVIEW

File: src/views/settings/CategorizationSettings.vue [hunk 1]
```diff
@@ -61,6 +66,33 @@ export default {
     restoreDefaultClasses: async function () {
       await this.$store.commit('categories/restoreDefaultClasses');
     },
+    exportClasses: function () {
+      console.log('Exporting categories...');
+
+      const text = JSON.stringify(this.classes_hierarchy);
```
Comment (ErikBjare) [inline src/views/settings/CategorizationSettings.vue:72]: I don't think `classes_hierarchy` is the right thing to export. Just export the list of categories, as they are laid out in `localStorage`.

**Comment by ErikBjare** [comment]
@johan-bjareholt I think the format is good, and just to be clear I expect the file/storage format to be:

```py
class Category:
    name: List[str]
    rule: Rule
    data: Dict | None
    id: int | None  # should probably be discarded/regenerated when exporting/importing

categories: List[Category]
```

**Comment by johan-bjareholt** [comment]
@ErikBjare When exporting, would it make most sense to export what's currently being edited or what's saved?

**Comment by ErikBjare** [comment]
@johan-bjareholt I thought a bit about that too, but I don't think we should support exporting unsaved changes. 

And on the flip side: we should maybe import everything as unsaved changes, to allow for review before saving?

**Comment by johan-bjareholt** [comment]
@ErikBjare Seems to work well now! Ready for review

**Comment by johan-bjareholt** [comment]
When exporting, it exports what's saved.
When importing, it imports as unsaved changes.

Format is { "categories": class dict }

**Review by ErikBjare — APPROVED** [review]

**Pull request merged**